source/docs/advanced/theme/sitemap.md

```diff
@@ -0,0 +1,172 @@
+
+## Generate a Sitemap
+
+Within Front-Commerce you can generate a sitemap by running the command `npm run sitemap` in your Skeleton. It will run a GraphQL query that will fetch all the routes registered in your application.
+
+The GraphQL query looks like this:
+
+```graphql
+query Sitemap {
+  sitemap {
+    baseUrl
+    nodes {
+      path
+      priority
+      changefreq
+      lastmod
+    }
+  }
+}
+```
+
+This query will be run for each store available in your `config/stores.js` file.
+
+If the request does not work, please make sure that the environment variable `FRONT_COMMERCE_SITEMAP_TOKEN` is available when running your script.
+
+The result of this query will then be transformed in xml files that are available in the `build/client/sitemaps` folder.
+
+> On your production website, we suggest that you execute this request every night through a cron to ensure that the sitemap is up to date.
+
+## Add your own routes in the sitemap
+
+By default Front-Commerce will return in the sitemap's nodes the nodes registered by the modules declared in your application. For instance, if in your `.front-commerce.js` you have registered `server/modules/magento2` in your `serverModules`, you should have categories, products and CMS pages available in your sitemap.
+
+However, if you need to setup your own routes, you will need to register your own nodes.
+
+### Add a static page
+
+If the pages you are adding to your sitemap are static and exist regardless of the state of your shop, you can add them by creating `config/hardcodedSitemap.js` in your module.
+
+```js
+module.exports = [
+  {
+    path: "/",
+    priority: 1,
+    changefreq: "daily",
+    lastmod: new Date("2017-06-01").toISOString()
+  }
+];
+```
+
+### Add dynamic pages
+
+However, if it relies on dynamic information (like an id or a slug in your URL), you will need to follow these two steps:
+
+- declare a GraphQL type as `Sitemapable`
+- register the pages by calling `loaders.Sitemap.registerNodesFetcher` in a module's `contextEnhancer`.
+
+Let's say that we have a FAQ and that each question have a route `/faq/:slug`. In order to do this, we will consider that we already created a GraphQL modules that:
+
+- declares the following types
+
+  ```graphql
+  # src/server/modules/faq/schema.gql
+  type FaqEntry {
+    slug: String
+    question: String
+    answer: String
+  }
+
+  extend type Query {
+    allFaq: [FaqEntry]
+    faq(slug: String!): FaqEntry
+  }
+  ```
+
+- declares a loader that allows to fetch all the FAQ entries and returns an array of object containing the slug, the question and the answer for each FAQ entry.
+  ```js
+  // src/server/modules/faq/loaders.js
+  const FaqLoader = () => {
+    return {
+      allFaq: () => {
+        return fetch("https://my-faq-service.example.org/all").then(
+          response => response.data
+        );
+      }
+    };
+  };
+  ```
+
```

Comment:
I feel like a transition is missing. While not very inspired, here is a proposal:

```suggestion

Here is how we could then include all pages in the sitemap.
```